### ISSUE TITLE
False Negative: Interrupted-exception swallowing is missed once the `catch` or `throws` structure is rewritten.

### ISSUE THREAD

**Issue by Carlson-JLQ** [google/error-prone#5635]
# False Negative: Interrupted-exception swallowing is missed once the `catch` or `throws` structure is rewritten.

Version
unknown in repository metadata

## Checker
- Checker id: `InterruptedExceptionSwallowed`
- Checker description: This checker detects catch blocks that catch a supertype of InterruptedException (like Exception or Throwable) without handling the interruption separately, and methods that can throw InterruptedException but declare a broader throws clause.

## Description of the false negative
`diff/output` marks 1 false negatives for `InterruptedExceptionSwallowed`. Across these samples, the benchmark is still exercising the same underlying bug pattern, but the implementation appears to lose track of it once the source deviates from one canonical shape. The misses read more like matching gaps than deliberate exclusions.

### `PosCase1.java`
Source: `output/sonarqube/InterruptedExceptionCheck/src/main/java/scensct/core/pos/PosCase1.java`

This is a concurrency-pattern sample where the same bug or non-bug is expressed through synchronization primitives rather than a single canonical syntax form. The risky pattern is still present in the code, so the absence of a report points to a matching gap rather than a benign variant. That is also how the benchmark classifies the sample: Catching InterruptedException without re-interrupting or rethrowing should be flagged as violation.

```java
// Catching InterruptedException without re-interrupting or rethrowing should be flagged as violation.
package scensct.core.pos;

public class PosCase1 {
    public static void main(String[] args) {
        try {
            Thread.sleep(1000); // Statement that can throw InterruptedException
        } catch (InterruptedException e) { // [REPORTED LINE]
            // Scenario 1: Catch InterruptedException, no re-interrupt, no rethrow
            System.out.println("Interrupted");
        }
    }
}
```

## Cause analysis
The rule is documented as follows: This checker detects catch blocks that catch a supertype of InterruptedException (like Exception or Throwable) without handling the interruption separately, and methods that can throw InterruptedException but declare a broader throws clause.

 The missed cases suggest that the implementation is narrower than the rule description and too dependent on one preferred AST shape.

Once the same behavior is expressed through a helper, an overload, a modifier such as `abstract` or `native`, a different comment position, or a slightly rearranged control-flow structure, the report disappears even though the benchmark still treats the case as in-scope.

For `InterruptedExceptionSwallowed`, the practical improvement would be to generalize the match so it follows the underlying behavior rather than one exact spelling of it.

## References
None known.
